runtime/context.go

```diff
@@ -137,3 +137,51 @@ func timeoutUnitToDuration(u uint8) (d time.Duration, ok bool) {
 	}
 	return
 }
+
+// isPermanentMessageHeader checks whether hdr belongs to the list of permenant
+// permenant request headers maintained by IANA. Any other headers are
+// classified as the user-specified metadata.
+// http://www.iana.org/assignments/message-headers/message-headers.xml
+func isPermanentHeader(hdr string) bool {
+	if hdr != "" && hdr[0] == ':' {
+		return true
+	}
+	switch hdr {
+	case
```

Comment:
Initially I was concerned that this would be slow because of a bunch of calls to strcmp. Turns out that golang builds a binary tree at compile time and uses that to search. Cool!

https://github.com/golang/go/issues/10000